parsl/dataflow/dflow.py

```diff
@@ -789,15 +788,21 @@ def add_executors(self, executors):
             if hasattr(executor, 'provider'):
                 if hasattr(executor.provider, 'script_dir'):
                     executor.provider.script_dir = os.path.join(self.run_dir, 'submit_scripts')
-                    if executor.provider.channel.script_dir is None:
-                        executor.provider.channel.script_dir = os.path.join(self.run_dir, 'submit_scripts')
-                        if not executor.provider.channel.isdir(self.run_dir):
-                            parent, child = pathlib.Path(self.run_dir).parts[-2:]
-                            remote_run_dir = os.path.join(parent, child)
-                            executor.provider.channel.script_dir = os.path.join(remote_run_dir, 'remote_submit_scripts')
-                            executor.provider.script_dir = os.path.join(self.run_dir, 'local_submit_scripts')
-                    executor.provider.channel.makedirs(executor.provider.channel.script_dir, exist_ok=True)
                     os.makedirs(executor.provider.script_dir, exist_ok=True)
+
+                    if hasattr(executor.provider, 'channels'):
+                        logger.debug("Creating script_dir across ad-hoc cluster")
```

Comment:
I think this functionality is a bit broken here, let me fix that.

Reply:
@benclifford Can you take a look at this bit again ?

Reply:
The logic for multiple channels and one channels looks (unnecessarily?) different: for example, a single channel provider can omit a script dir but a multi-channel provider cannot omit a script dir. Maybe pull the original block into a helper function so that it can either be invoked once on `channel` or invoked multiple times on each element of `channels`?

Reply:
Done.